### DISCUSSION TITLE
ID monster

### DISCUSSION THREAD

**Discussion by TienDat2k** [aieuo/Mineflow#81] [Q&A]
sorry to bother again. I have a question to ask you because it is very important to me. I tried to populate the monster id in "active entity id" which is the event trigger but it doesn't work. I did a search through 'pocketmine' which only shows the id as a specific name and not the id, but when I retype it, it still doesn't work. How can I properly fill in the monster's Id in the active condition when I don't know its id?

**Comment by aieuo** [comment]
Could you send me a screenshot of the form with the "active entity id" displayed? I'm not sure where it is displayed...

**Reply by TienDat2k** [reply]
![Screenshot_2021-08-26-10-32-50-250_com.mojang.minecraftpe.jpg](https://user-images.githubusercontent.com/88370821/130896741-0da71e59-a029-409b-a568-cde3dd20372a.jpg)

**Reply by TienDat2k** [reply]
![Screenshot_2021-08-26-10-33-03-845_com.mojang.minecraftpe.jpg](https://user-images.githubusercontent.com/88370821/130896788-14e52b20-0daf-4a0e-8d6a-9d0786c44a98.jpg)

**Reply by TienDat2k** [reply]
![Screenshot_2021-08-26-10-33-20-279_com.mojang.minecraftpe.jpg](https://user-images.githubusercontent.com/88370821/130896821-d8cabc7f-41d3-40fc-ab63-498d0289aa95.jpg)

**Reply by TienDat2k** [reply]
I don't know exactly what id to enter because there is no obvious id for the monster in pocketmine

**Reply by aieuo** [reply]
Thanks, I got it.  
It is the entity id inside the server and you can get it from variables like `{target.id}`.  
Actions "If the target is alive and online" and "Check if an entity with {target.id} ID is alive and online" behave exactly the same.

**Reply by TienDat2k** [reply]
What I mean here is that I want to uniquely identify an entity in my server monster list. The reason is because when I use the damage value it works with all the monster list in my server so I want to identify exactly a single monster that is allowed to use through the condition . Maybe i am doing wrong or there is problem with id, i looked for id list in pocketmine but no id with monster. Please help me with this problem

**Reply by aieuo** [reply]
There is currently no way to detect such an entity. I will add that feature.

**Reply by TienDat2k** [reply]
If you don't mind, can you give me some examples of how to use the form? I hope one day you will reappear on youtube with more interesting examples in the future.

**Reply by aieuo** [reply]
sure! I'll create videos of form examples.

**Reply by TienDat2k** [reply]
wonderful

**Reply by aieuo** [reply]
> What I mean here is that I want to uniquely identify an entity in my server monster list. The reason is because when I use the damage value it works with all the monster list in my server so I want to identify exactly a single monster that is allowed to use through the condition . Maybe i am doing wrong or there is problem with id, i looked for id list in pocketmine but no id with monster. Please help me with this problem

I added a variable property to get entity type in [this build](https://poggit.pmmp.io/ci/aieuo/Mineflow/Mineflow/dev:201)  
To check if an entity is a zombie, use it like this
![image](https://user-images.githubusercontent.com/35859665/130969261-2d6bb7c5-8c75-4c23-8cf1-7ee048fe23d3.png)

**Reply by TienDat2k** [reply]
Thank you. I'll try it right away, if there's anything wrong I'll report back to you later.

**Reply by TienDat2k** [reply]
wonderful. it works great

**Reply by TienDat2k** [reply]
there doesn't seem to be a variable or action when the entity dies. I want to add a specific effect or item to the player when they kill a monster but I don't see a variable to meet that condition

**Reply by aieuo** [reply]
There is a `When an entity or player dies` event trigger. If it is not showing, go to `Settings > Event` and make sure it is enabled.

**Reply by TienDat2k** [reply]
Thank you for answering my questions in the past, you may find it a bit annoying but still hope you will still answer my problems in the future, sincerely thank you. And now I have a new problem again (;;;・_・).How can i add action to the monster. Eg. When the monster hits a block it can automatically jump over or if hit it will be knocked back "knockback", okay? .Hope you can help me

**Reply by TienDat2k** [reply]
![Screenshot_2021-08-26-23-45-25-041_com mojang minecraftpe](https://user-images.githubusercontent.com/88370821/131146813-98821b35-ae12-4665-97de-56565d93c805.jpg)

**Reply by TienDat2k** [reply]
しばらくして、私は新しい問題を発見しました。 10体以上のモンスターを召喚して同時に攻撃すると、すべてのモンスターが攻撃しますが、デフォルトではダメージ量は1ですが、これはバグですか？または、このフィールドの数式を作成する必要がありますか？変数でモンスターまたはプレイヤーの健康状態を確認するにはどうすればよいですか？モンスターや将来のプレイヤーのためにエキサイティングなトリガーを促進するために必要です。

**Reply by aieuo** [reply]
No problem!

> When the monster hits a block it can automatically jump over.

Try using the `If`, `Get the coordinates of an entity's side.`, and `Get a block with specified coordinates` actions to see if the block forward of the entity's feet is air, and let the entity jump when it's not air

> knockback

There is no action for knockback now. I'll add it.

**Reply by aieuo** [reply]
> しばらくして、私は新しい問題を発見しました。 10体以上のモンスターを召喚して同時に攻撃すると、すべてのモンスターが攻撃しますが、デフォルトではダメージ量は1ですが、これはバグですか？または、このフィールドの数式を作成する必要がありますか？変数でモンスターまたはプレイヤーの健康状態を確認するにはどうすればよいですか？モンスターや将来のプレイヤーのためにエキサイティングなトリガーを促進するために必要です。

I think it's probably due to the cool down after taking damage.
To check the entity's health, you can use `{entity.health}` and `{entity.maxHealth}` variable property.

**Reply by TienDat2k** [reply]
> Có một `When an entity or player dies`kích hoạt sự kiện. Nếu nó không hiển thị, hãy truy cập `Settings > Event`và đảm bảo rằng nó đã được bật.

can you give me a specific example, i tried to do it but failed many times and i don't know how.I don't know how to apply the variable correctly.
